### ISSUE TITLE
[Linux] batch multiple process info in one shot

### ISSUE THREAD

**Issue by giampaolo** [giampaolo/psutil#2454]
This is a reminder ticket to investigate some links that @iglendd shared in https://github.com/giampaolo/psutil/issues/2366: 

* https://criu.org/Task-diag
* https://avagin.github.io/how-fast-is-procfs.html
* https://patchwork.kernel.org/project/linux-fsdevel/cover/20200810145852.9330-1-elubarsky.linux@gmail.com/#23542875
* https://lwn.net/Articles/685791/

I didn't investigate any of these links, so I'm not sure if this is even possible, but more or less the idea here is to fetch info about multiple process in one shot, and integrate this in `psutil.process_iter()`. Or to put it another way, there may (or rather will) be an alternative method to reading pseudo-files from `/proc`, which is quite expensive on Linux (in fact it may be the slowest implementation across all supported platforms).

**Comment by iglendd** [comment]
@giampaolo I did investigated it a bit after I have shared with you. It appears that in theory there are a few alternative method had been discussed a while ago but I think they have never materialized in the main Linux releases (despite obvious shortcomings in the /proc API calls. I could be wrong but that is my impression. :(.

**Comment by giampaolo** [comment]
No, unfortunately my impression is that parsing files in /proc is the only way to query processes on Linux. Which is error prone and often slow, but it is what it is. :-(
